datafusion/core/src/physical_plan/limit.rs

```diff
@@ -462,7 +462,8 @@ impl LimitStream {
                 .iter()
                 .map(|col| col.slice(0, col.len().min(batch_rows)))
                 .collect();
-            Some(RecordBatch::try_new(batch.schema(), limited_columns).unwrap())
+            let options = RecordBatchOptions::new().with_row_count(Option::from(batch_rows));
```

Comment:
Thanks @kazuyukitanimura 
extending call `RecordBatch` to be called with options gives more flexibility

Reply:
I agree it is nice